src/pages/Markets/hooks/usePortalsAssetsQuery.ts

```diff
@@ -68,11 +68,11 @@ export const usePortalsAssetsQuery = ({ chainIds }: { chainIds: ChainId[] | unde
                 byId: { [assetId]: asset },
               }),
             )
-
-            // and its market-data since it may or may not be missing
-            dispatch(marketApi.endpoints.findByAssetId.initiate(assetId))
           }
 
+          // and fetch its market-data since it may or may not be missing
+          dispatch(marketApi.endpoints.findByAssetId.initiate(assetId))
+
```

Comment:
This was previously fetching only *if* the assetId doesn't yet exist in assets slice